src/custom-client-scripts/client-script.js

```diff
@@ -0,0 +1,106 @@
+import { readFile, fsObjectExists } from '../utils/promisified-functions';
+import { GeneralError } from '../errors/runtime';
+import { RUNTIME_ERRORS } from '../errors/types';
+import resolvePathRelativelyCwd from '../utils/resolve-path-relatively-cwd';
+import { RequestFilterRule, generateUniqueId } from 'testcafe-hammerhead';
+
+const BEAUTIFY_REGEXP = /[/.:\s\\]/g;
+const BEAUTIFY_CHAR   = '_';
+
+const EMPTY_CONTENT_STR      = '{ content: <empty> }';
+const CONTENT_STR_MAX_LENGTH = 30;
+const CONTENT_ELLIPSIS_STR   = '...';
+
+const URL_UNIQUE_PART_LENGTH = 7;
+
+export default class ClientScript {
+    constructor (init) {
+        this.init    = init || null;
+        this.url     = generateUniqueId(URL_UNIQUE_PART_LENGTH);
+        this.content = '';
+        this.path    = null;
+        this.module  = null;
+        this.page    = RequestFilterRule.ANY;
+    }
+
+    async _resolveAsRelativePath (path) {
+        const resolvedPath       = resolvePathRelativelyCwd(path);
+        const resolvedPathExists = await fsObjectExists(resolvedPath);
+
+        return resolvedPathExists ? resolvedPath : null;
+    }
+
+    async _loadFromPath (path) {
+        try {
+            const resolvedPath = await this._resolveAsRelativePath(path);
```

Comment:
Here we check that file exists and then do `readFile`
Can we do both these operations at once?